src/pages/main/ui/datasets-list/datasets-list.utils.tsx

```diff
@@ -0,0 +1,18 @@
+import dirinfoStore from '@store/dirinfo'
+import { DatasetsListItem } from '@pages/main/ui/datasets-list/datasets-list-item'
+import { IDirInfoDatasetDescriptor } from '@service-providers/vault-level/vault-level.interface'
+
+export const datasetNameByKey =
+  (level: number = 0) =>
+  // eslint-disable-next-line react/display-name
+  (key: string) => {
```

Comment:
what's the problem to just make a function with 2 params? Avoid closures when it's not necessary